src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java

```diff
@@ -138,45 +292,311 @@ private void renderImageRegion(Message<String> message) {
+            }
+        });
+
+        timer.observeDuration();
+        return future;
+    }
+
+    /**
+     * Get cached {@link Pixels} metadata if available or retrieve it from the
+     * server.
+     * @param imageRegionCtx request context
+     * @param request OMERO request based on the current context
+     * @param requestHandler OMERO image region request handler
+     * @return Future which will be completed with the {@link Pixels} metadata.
+     */
+    private Future<Pixels> getPixels(
+            ImageRegionCtx imageRegionCtx, Supplier<OmeroRequest> request,
+            ImageRegionRequestHandler requestHandler) {
+        Summary.Timer timer = getPixelsSummary.startTimer();
+        Future<Pixels> future = Future.future();
+
+        String key = String.format("%s:Image:%d",
+                Pixels.class.getName(), imageRegionCtx.imageId);
+        vertx.eventBus().<byte[]>send(
+                RedisCacheVerticle.REDIS_CACHE_GET_EVENT, key, result -> {
+            try {
+                final byte[] serialized = result.succeeded()?
+                        result.result().body() : null;
+                if (serialized != null) {
+                    Future<Boolean> step1 =
+                            canRead(imageRegionCtx, request, requestHandler);
+
+                    step1.compose(canRead -> {
+                        if (canRead) {
+                            try (ObjectInputStream oos = new ObjectInputStream(
+                                    new ByteArrayInputStream(serialized))) {
+                                log.info("Cache HIT {}", key);
+                                pixelsCacheHit.inc();
+                                future.complete((Pixels) oos.readObject());
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
+                            }
+                        } else {
+                            try {
+                                log.info("Cache MISS {}", key);
+                                pixelsCacheMiss.inc();
+                                future.complete(loadPixels(
+                                        request, requestHandler, key));
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
+                    }, future);
+                } else {
+                    try {
+                        log.info("Cache MISS {}", key);
+                        pixelsCacheMiss.inc();
+                        future.complete(loadPixels(
+                                request, requestHandler, key));
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            } catch (Exception e) {
+                future.fail(e);
+            }
+        });
+
+        timer.observeDuration();
```

Comment:
With all the async code above, observing the duration here won't tell us anything, it will end up being 0ms.